### PR TITLE
[fix](sequence column) fix update fail on nereids planner

### PR DESCRIPTION
**Pull request by zhannngchen**
## Proposed changes

Issue Number: close #xxx

1. if we set `enable_fallback_to_original_planner` to false, the `UPDATE` sql in regression case `test_unique_table_sequence` will fail due to：

    ```
    Table test_uniq_sequence has sequence column, need to specify the sequence column
    ```
    The bug is introduced by #27780 
2. fix insert fail in transaction mode, which is introduced by #27947 
3. add cases with session variable `enable_fallback_to_original_planner` false

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhannngchen** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java [hunk 1]
```diff
@@ -167,7 +168,9 @@ public List<Rule> buildRules() {
                                 }
                             }
 
-                            if (!haveInputSeqCol && !isPartialUpdate) {
+                            if (!haveInputSeqCol && !isPartialUpdate && (
+                                    boundSink.getDmlCommandType() != DMLCommandType.UPDATE
+                                            && boundSink.getDmlCommandType() != DMLCommandType.DELETE)) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java:173]: could we add a help function in DMLCommandType for easy use?

Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java:173]: please add some comment to explain why to chcek type != update and delete

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/Command.java [hunk 2]
```diff
@@ -39,6 +39,17 @@
  */
 public abstract class Command extends AbstractPlan implements LogicalPlan, BlockFuncDepsPropagation {
 
+    /**
+     * Type of DML Command
+     */
+    public enum DMLCommandType {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/Command.java:45]: do not add thie enum here, create a new file under command dir

**Comment by zhannngchen** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java (hunk 1 above)
Comment (zhannngchen) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java:177]: sounds not necessary? only 4 types

Comment (zhannngchen) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindSink.java:177]: done

**Comment by zhannngchen** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by zhannngchen** [comment]
run clickbench-new

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**